### PR TITLE
CSP Violation: form input onfocus

### PR DESCRIPTION
**Pull request by chdorner**
I'm not sure if this is the correct way to handle this, so I've held off writing tests.

One thing I noticed is that this doesn't work right after a page load, presumably because the `focus` event is fired before all the JS code is loaded (as described [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-autofocus)).

### CODE REVIEW

**Comment by chdorner** [comment]
Any thoughts on this would be appreciated, this is all a bit out of my depth at the moment.

File: h/static/scripts/form-select-onfocus-controller.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+function FormSelectOnFocusController(element) {
+  var self = this;
+
+  this._formInputs = element.querySelectorAll('form input.js-select-onfocus');
```
Comment (robertknight) [inline h/static/scripts/form-select-onfocus-controller.js]: Selectors should be minimally specific, so just `.js-select-onfocus` would be fine here.

File: h/static/scripts/site.js [hunk 2]
```diff
@@ -24,6 +25,10 @@ page('/groups/new', function () {
   new CreateGroupFormController(document.body);
 });
 
+page('/login', function() {
+  new FormSelectOnFocusController(document.body);
```
Comment (robertknight) [inline h/static/scripts/site.js]: This issue and the other one just means there should be a `'use strict';` at the top of the file.

**Comment by robertknight** [comment]
> One thing I noticed is that this doesn't work right after a page load,
> presumably because the focus event is fired before all the JS code is loaded

That might indeed be the case. In which case you can use `document.activeElement` to test whether the element has focus at the point when the JS code loads and select it if so.

File: h/static/scripts/form-select-onfocus-controller.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+function FormSelectOnFocusController(element) {
+  this._elements = element.querySelectorAll('.js-select-onfocus');
+
+  this._onfocusHandler = function(event) {
+    event.target.select();
+  };
+
+  for (var i = 0; i < this._elements.length; i++) {
+    var element = this._elements[i];
```
Comment (chdorner) [inline h/static/scripts/form-select-onfocus-controller.js]: This is fixed now.

File: h/static/scripts/site.js (hunk 2 above)
Comment (chdorner) [inline h/static/scripts/site.js]: This is fixed now.

**Comment by chdorner** [comment]
Thanks @robertknight. I've fixed these issues, and added tests. This is now ready for a final review.

**Comment by robertknight** [comment]
LGTM

**Pull request merged**